plugins/okta/komand_okta/connection/connection.py

```diff
@@ -29,3 +31,42 @@ def test(self):
             return {"success": True}
         except PluginException as error:
             raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=error.data)
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Okta")
+        return_message = "The connection test to Okta was unsuccessful \n"
+        try:
+            _ = self.api_client.list_events("")
+
+            message = "The connection test to Okta was successful"
+            self.logger.info(message)
+            return {"success": True}, message
+
+        except ApiException as error:
+
+            cause_msg = f"The failed connection test to Okta was caused by: '{error.cause}'"
+            return_message += f"{cause_msg}\n"
+            self.logger.info(cause_msg)
+
+            self.logger.info(error.assistance)
+            return_message += f"{error.assistance}\n"
+
+            self.logger.error(error)
+            raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=return_message)
+
+        # This catches if the user enters a URL in the correct format but is unreachable
+        except ConnectionError as error:
+            self.logger.info(error)
+            raise ConnectionTestException(
+                cause="The URL provided in the connection is unreachable. Please ensure you are providing a valid URL when trying to connect.",
+                assistance="Please verify the URL specified is valid and reachable before attempting to connect again.",
+                data="Please ensure the URL specified can be reached and is valid before trying to reconnect.",
+            )
+
+        except ConnectionResetError as error:
+            self.logger.info("Catching egress rules for connection", error)
```

Comment:
I think you've slightly mistyped this logger? Should 'error' be passed into the logging string? Can we check during testing this logs the URL that we failed to connect to as well please